File: VB/Q154615/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports System.IO
Imports System.Data.Common
Imports Q154615.nwindDataSetTableAdapters

Namespace Q154615
	Partial Public Class Form1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
			PopulateSelectDataSourceComboBox()
			PopulateTables()
		End Sub

		Private Sub PopulateSelectDataSourceComboBox()
			For Each table As DataTable In nwindDataSet.Tables
				cbeSelectDataSource.Properties.Items.Add(table.TableName)
			Next table
		End Sub

		Private Sub PopulateTables()
			categoriesTableAdapter.Fill(nwindDataSet.Categories)
			productsTableAdapter.Fill(nwindDataSet.Products)
			order_DetailsTableAdapter.Fill(nwindDataSet.Order_Details)
			ordersTableAdapter.Fill(nwindDataSet.Orders)
		End Sub

		Private Sub OnSelectedDataSourceChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbeSelectDataSource.SelectedIndexChanged
			DoChangeDataSource((CType(sender, BaseEdit)).Text)
		End Sub

		Private Sub DoChangeDataSource(ByVal dataMember As String)
			If (Not String.IsNullOrEmpty(nwindBindingSource.DataMember)) Then
				layout.SaveLayoutToXml(GetLayoutPath(nwindBindingSource.DataMember))
			End If
			layout.Clear()
			nwindBindingSource.DataMember = dataMember
			layout.RetrieveFields()
			Dim layoutPath As String = GetLayoutPath(dataMember)
			If File.Exists(layoutPath) Then
				layout.RestoreLayoutFromXml(layoutPath)
			Else
				layout.ShowCustomizationForm()
			End If
		End Sub

		Private Function GetLayoutPath(ByVal dataMember As String) As String
			Return String.Format("..\..\{0}_layout.xml", dataMember)
		End Function
	End Class
End Namespace

File: CS/Q154615/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using System.IO;
using System.Data.Common;
using Q154615.nwindDataSetTableAdapters;

namespace Q154615 {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            PopulateSelectDataSourceComboBox();
            PopulateTables();
        }

        private void PopulateSelectDataSourceComboBox() {
            foreach (DataTable table in nwindDataSet.Tables)
                cbeSelectDataSource.Properties.Items.Add(table.TableName);
        }

        private void PopulateTables() {
            categoriesTableAdapter.Fill(nwindDataSet.Categories);
            productsTableAdapter.Fill(nwindDataSet.Products);
            order_DetailsTableAdapter.Fill(nwindDataSet.Order_Details);
            ordersTableAdapter.Fill(nwindDataSet.Orders);
        }

        private void OnSelectedDataSourceChanged(object sender, EventArgs e) {
            DoChangeDataSource(((BaseEdit)sender).Text);
        }

        private void DoChangeDataSource(string dataMember) {
            if (!string.IsNullOrEmpty(nwindBindingSource.DataMember))
                layout.SaveLayoutToXml(GetLayoutPath(nwindBindingSource.DataMember));
            layout.Clear(true, true);
            nwindBindingSource.DataMember = dataMember;
            layout.RetrieveFields();
            string layoutPath = GetLayoutPath(dataMember);
            if (File.Exists(layoutPath)) layout.RestoreLayoutFromXml(layoutPath);
            else layout.ShowCustomizationForm();
        }

        private string GetLayoutPath(string dataMember) { 
            return string.Format(@"..\..\{0}_layout.xml", dataMember); 
        }
    }
}